jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/Spatial.java

```diff
@@ -69,13 +68,22 @@ default Rectangle2D getUnion(Rectangle2D rect, Point2D p) {
     return new Rectangle2D.Double(left, top, right - left, bottom - top);
   }
 
+  default Rectangle2D getUnion(Rectangle2D rect, double x, double y) {
```

Comment:
Technically this is not a union; you're expanding the input Rectangle (if needed) so that it will include (x,y).  (A union of a rectangle and a point is...a rectangle, and maybe a point.)   Calling this a "union" is confusing.

Maybe `expandRectangleToInclude()`?  `getRectangleIncluding()`?

Note that if you should be using the new JUNG `Rectangle` class instead, maybe this could be a method on that class (similar to the `offsetWith()` method I proposed for `Point`).